pkg/volume/host_path/host_path.go

```diff
@@ -103,6 +104,12 @@ func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volum
 	if err != nil {
 		return nil, err
 	}
+
+	err = validation.ValidatePathNoBacksteps(hostPathVolumeSource.Path)
```

Comment:
Erroring in `SetUp(...)` would be better, since that will automatically result in a mount failure and the generation of an event associated with the pod.